classes/output/core_renderer.php

```diff
@@ -463,4 +463,63 @@ public function block(\block_contents $bc, $region) {
 
         return $this->render_from_template('core/block', $context);
     }
+
+    /**
+     * Allow plugins to provide some content to be rendered in the navbar.
+     * The plugin must define a PLUGIN_render_navbar_output function that returns
+     * the HTML they wish to add to the navbar.
+     *
+     * @return string HTML for the navbar
+     */
+    public function navbar_plugin_output() {
+        $output = parent::navbar_plugin_output();
+        return $this->get_favourites() . $output;
+    }
+
+    /**
+     * Function that fetches all favorite courses, and renders them as a popover menu.
+     *
+     * @return string HTML to display the main header.
+     */
+    private function get_favourites() {
```

Comment:
-> this should be moved into locallib.php, keeping core_renderer.php free of new features

Reply:
Thanks for the feedback, fixed too :)